src/Compilers/CSharp/Portable/CSharpExtensions.cs

```diff
@@ -317,6 +317,22 @@ internal static Syntax.InternalSyntax.DirectiveStack ApplyDirectives(this Syntax
             return builder.ToList();
         }
 
+        #region ITypeSymbol
+
+        /// <summary>
+        /// Returns true iff the supplied type is sbyte, byte, short, ushort, int, uint,
+        /// long, ulong, bool, char, string, or an enum-type, or if it is the nullable type
+        /// corresponding to one of those types. These types were permitted as the governing
+        /// type of a switch statement in C# 6.
+        /// </summary>
+        public static bool IsValidV6SwitchGoverningType(this ITypeSymbol type)
```

Comment:
I don't think there is a simpler name. For example, it includes enum types.

The implementation is simple enough that I think it can just be coped into the E&C code. I'll place the implementation in the issue for this, and close this PR.

---
In reply to: [77084316](https://github.com/dotnet/roslyn/pull/13498#discussion_r77084316) [](ancestors = 77084316)